### PR TITLE
feat: ApisixRoute v2alpha1

### PR DESCRIPTION
**Pull request by tokers**
Please answer these questions before submitting a pull request

- Why submit this pull request?
- [ ] Bugfix
- [x] New feature provided
- [ ] Improve performance
- [ ] Backport patches

- Related issues

#248 
___

### New feature or improvement
- Describe the details and related test reports.
This  PR implemented the ApisixRoute.apisix.apache.org/v2/alpha1 resource. Generate the required listers, informers for it.

### CODE REVIEW

**Comment by tokers** [comment]
@gxthrj Please take a look when you have time.

**Comment by gxthrj** [comment]
Conflicting files

File: pkg/ingress/controller/controller.go [hunk 1]
```diff
@@ -338,17 +340,6 @@ type Api6Controller struct {
 	Stop                      <-chan struct{}
 }
 
-func (api6 *Api6Controller) ApisixRoute(controller *Controller) {
```
Comment (gxthrj) [inline pkg/ingress/controller/controller.go:341]: We discussed earlier that we should keep the v1 version for a while, declare in the document "deprecated", and delete the logic of v1 in version 1.0.

Comment (tokers) [reply pkg/ingress/controller/controller.go:341]: Sorry I forgot to set this PR to draft, the v1 ApisixRoute will be kept of course.

File: test/e2e/ingress/resourcepushing.go [hunk 2]
```diff
@@ -64,19 +76,21 @@ spec:
...
+      - httpbin.com
       paths:
-      - backend:
-          serviceName: %s
-          servicePort: %d
-        path: /ip
+      - /ip
+    backend:
+      serviceName: %s
+      servicePort: %d
```
Comment (gxthrj) [inline test/e2e/ingress/resourcepushing.go:93]: Need more test case with v2alpha1. We may add more in future.

**Review by gxthrj — APPROVED** [review]

File: test/e2e/ingress/resourcepushing.go (hunk 2 above)
Comment (tokers) [reply test/e2e/ingress/resourcepushing.go:93]: OK.

**Pull request merged**